src/shogun/mathematics/graph/CMakeLists.txt

```diff
@@ -1,29 +1,62 @@
-set(SHOGUN_GRAPH_HEADERS
-	Allocator.h
-	Graph.h
-	LinalgNodes.h
-	OperationImplementation.h
-	Shape.h
-	Tensor.h
-	Types.h
-	ops/Add.h
-	ops/Input.h
+set(SHOGUN_CORE_GRAPH_HEADERS
+	${CMAKE_CURRENT_SOURCE_DIR}/Allocator.h
+	${CMAKE_CURRENT_SOURCE_DIR}/Graph.h
+	${CMAKE_CURRENT_SOURCE_DIR}/LinalgNodes.h
+	${CMAKE_CURRENT_SOURCE_DIR}/OperatorImplementation.h
+	${CMAKE_CURRENT_SOURCE_DIR}/Shape.h
+	${CMAKE_CURRENT_SOURCE_DIR}/Tensor.h
+	${CMAKE_CURRENT_SOURCE_DIR}/Types.h
+	${CMAKE_CURRENT_SOURCE_DIR}/operator_list.h
+	${CMAKE_CURRENT_SOURCE_DIR}/ops/Add.h
+	${CMAKE_CURRENT_SOURCE_DIR}/ops/Input.h
 	)
 
-set(SHOGUN_GRAPH_SOURCES
-	Graph.cpp
-	Tensor.cpp
-	ops/Add.cpp
-	ops/Input.cpp
+set(SHOGUN_CORE_GRAPH_SOURCES
+	${CMAKE_CURRENT_SOURCE_DIR}/Graph.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/Tensor.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/operator_list.cpp
 	)
 
-add_library(shogun-graph SHARED ${SHOGUN_GRAPH_HEADERS} ${SHOGUN_GRAPH_SOURCES} ${SHOGUN_CORE_HEADERS} ${CMAKE_BINARY_DIR}/src/shogun/lib/config.h)
+set(SHOGUN_GRAPH_IMPLEMENTATION_HEADERS
+	${CMAKE_CURRENT_SOURCE_DIR}/ops/shogun/Add.h
+	${CMAKE_CURRENT_SOURCE_DIR}/ops/shogun/Input.h
+	)
+set(SHOGUN_GRAPH_IMPLEMENTATION_SOURCES
+	${CMAKE_CURRENT_SOURCE_DIR}/ops/shogun/Operators.cpp
+	)
+
+# set(NGRAPH_GRAPH_IMPLEMENTATION_HEADERS
+# 	ops/ngraph/Add.h
+# 	)
+# set(NGRAPH_GRAPH_IMPLEMENTATION_SOURCES
+# 	ops/ngraph/Add.cpp
+# 	)
+
+add_library(shogun-graph SHARED ${SHOGUN_GRAPH_HEADERS} 
+	${SHOGUN_GRAPH_SOURCES} 
+	${SHOGUN_CORE_HEADERS} 
```

Comment:
do we need this? shouldn't a `target_link_libraries(shogun-graph shogun-core)` solve this?

Reply:
yup it does, fixed this